graph_wrap/django_rest_framework/api_transformer.py

```diff
@@ -33,7 +33,7 @@ def non_root_types(self):
         non_root_types = []
         for nested in self._nested_serializers:
             nested_transformed = SerializerTransformer(
-                nested, self.type_mapping).graphene_object_type()
+                nested, self.type_mapping, nested=True).graphene_object_type()
             non_root_types.append(nested_transformed)
```

Comment:
Unfortunately I don't think this can work. By doing this we'll treat any serializer embedded inside another serializer the same way we treat the dynamically created DRF `NestedSerializer` (dynamically created when a depth is set). 

The reason I want to treat them differently is because if we don't, the resulting schema becomes very confusing and can have field name collisions. The choice of naming for fields from NestedSerializers is documented here: https://github.com/PaulGilmartin/graph_wrap#understanding-the-schema. Here's a bit more on that: taking the `PostSerializer` as an example:

```
class PostSerializer(serializers.ModelSerializer):
    written_by = WrittenBySerializer(source='author')
    author = serializers.HyperlinkedRelatedField(
        view_name='author-detail',
        read_only=True
    )

    class Meta:
        model = Post
        depth = 3
        fields = [
            'written_by',
            'author',
            'content',
            'author',
            'date',
            'rating',
            'files',
        ]

```


the schema looks like this:

```
type post_type {
  written_by: author_type!
  content: String!
  date: String!
  files: [post__files_type]!
  rating: String
}
```
Notice that the `written_by` field is defined by `author_type` - this naming convention comes from the fact that written_by is defined by a custom serializer, `written_by = WrittenBySerializer(source='author')`.

Notice that `files` is defined by `post__files_type` - this naming convention comes from the fact that the file data is generated dynamically by a NestedSerializer (by virtue of the depth argument) - it points to the fact that there's no custom serializer defining the files field.
So now suppose we abandoned that naming convention `post__files_type` and simply called it `files_type`. This is when we run into problems. The schema of this `files_type` would still look the same:
```
type files_type {
  id: Int!
  name: String!
  content_type: String
  size: Int
}
```
- this schema is decided purely by how NestedSerializer works in DRF. But then suppose someone introduces a `MediaSerializer` into the code base which has say only the `size` field. If it was then used in some other serializer like this
```
files = MediaSerializer()
```
the schema from that type would look like this
```
type files_type {
  size: Int
}
```
which would conflict with the above.
Sorry for long rambling message! Let me know if anything needs clarification

Reply:
Thanks so much for your detailed response! I'll admit I'm still struggling to get my head around exactly what's going on with the schema generation. I probably put the cart before the horse here: I'm going to close out my two PRs and open a more detailed issue citing my use case and the foremost problem I'm running into, so that we can work from a common frame of reference. Thanks again!